isolate_example/lib/page_one.dart

```diff
@@ -12,20 +12,115 @@
 // limitations under the License.
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
-class PerformancePage extends StatelessWidget {
+class PerformancePage extends StatefulWidget {
+  @override
+  _PerformancePageState createState() => _PerformancePageState();
+}
+
+class _PerformancePageState extends State<PerformancePage> {
+  Future<void> computeFuture = Future.value();
+
   @override
   Widget build(BuildContext context) {
     return Center(
       child: Column(
         mainAxisSize: MainAxisSize.min,
         children: <Widget>[
           SmoothAnimationWidget(),
+          Container(
+            alignment: Alignment.bottomCenter,
+            padding: EdgeInsets.only(top: 150.0),
+            child: Column(
+              children: <Widget>[
+                FutureBuilder<void>(
+                  future: computeFuture,
+                  builder: (BuildContext context, AsyncSnapshot snapshot) {
+                    return RaisedButton(
+                      child: const Text('Compute on Main'),
+                      elevation: 8.0,
+                      onPressed: mainIsolateCallBack(context, snapshot),
+                    );
+                  },
+                ),
+                FutureBuilder<void>(
+                  future: computeFuture,
+                  builder: (BuildContext context, AsyncSnapshot snapshot) {
+                    return RaisedButton(
+                      child: const Text('Compute on Secondary'),
+                      elevation: 8.0,
+                      onPressed: secondaryIsolateCallBack(context, snapshot),
+                    );
+                  },
+                ),
+              ],
+            ),
+          ),
         ],
       ),
     );
   }
+
+  VoidCallback mainIsolateCallBack(
+      BuildContext context, AsyncSnapshot snapshot) {
+    if (snapshot.connectionState == ConnectionState.done) {
+      return () {
+        setState(() {
+          computeFuture = computeOnMainIsolate()
+            ..then((_) {
+              final snackBar1 = SnackBar(
+                content: Text('Main Isolate Done!'),
+              );
+              Scaffold.of(context).showSnackBar(snackBar1);
+            });
+        });
+      };
+    } else {
+      return null;
+    }
+  }
+
+  VoidCallback secondaryIsolateCallBack(
+      BuildContext context, AsyncSnapshot snapshot) {
+    if (snapshot.connectionState == ConnectionState.done) {
+      return () {
+        setState(() {
+          computeFuture = computeOnSecondaryIsolate()
+            ..then((_) {
+              final snackBar1 = SnackBar(
+                content: Text('Secondary Isolate Done!'),
+              );
+              Scaffold.of(context).showSnackBar(snackBar1);
+            });
+        });
+      };
+    } else {
+      return null;
+    }
+  }
+}
+
+Future<void> computeOnMainIsolate() async {
+  await Future.delayed(Duration(milliseconds: 200), () => fib(45));
```

Comment:
Add a comment here explaining why you're using a delay of 200ms, since it may not be obvious to most readers. Something simple like "Adding a delay before the start of the calculation gives Flutter the change to rebuild and redraw the screen first" is fine.